yt_dlp/extractor/nytimes.py

```diff
@@ -59,204 +51,546 @@ def get_file_size(file_size):
+                  ) {
+                    renditions {
+                      name
+                      width
+                      height
+                      url
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+      promotionalHeadline
+      promotionalMedia {
+        ... on Image {
+          crops(
+            cropNames: [
+              SMALL_SQUARE
+              MEDIUM_SQUARE
+              SIXTEEN_BY_NINE
+              THREE_BY_TWO
+              TWO_BY_THREE
+              FLEXIBLE
+            ]
+          ) {
+            name
+            renditions {
+              name
+              width
+              height
+              url
+            }
+          }
+        }
+      }
+      promotionalSummary
+      related {
+        ... on Article {
+          promotionalHeadline
+          url
+          sourceId
+        }
+      }
+      renditions {
+        type
+        width
+        height
+        url
+        bitrate
+      }
+      section {
+        name
+      }
+      shortUrl
+      sourceId
+      subsection {
+        name
+      }
+      summary
+      timesTags {
+        __typename
+        displayName
+        isAdvertisingBrandSensitive
+        vernacular
+      }
+      url
+    }
+  }
+}'''
+
+    def _build_playlist(self, media_items):
+        entries = []
+        for media_id in media_items:
+            json_obj = traverse_obj(self._json_from_graphql(media_id) or {}, ('data', 'video'))
+
+            title = json_obj.get('promotionalHeadline') or media_id
+            description = json_obj.get('summary')
+            creators = ', '.join(
+                [remove_start(creator, 'By ') for creator in traverse_obj(
+                    json_obj, ('bylines', ..., 'renderedRepresentation'))])
+            duration = int_or_none(json_obj.get('duration'))
+            media_content = json_obj.get('renditions')
+            formats, subtitles = self._extract_media_from_json(media_id, media_content)
+
+            thumbnails = []
+            for image in traverse_obj(json_obj, ('promotionalMedia', 'crops', ..., 'renditions', ...)):
+                image_url = image.get('url')
+                if not url_or_none(image_url):
+                    continue
+                thumbnails.append({
+                    'url': image_url,
+                    'width': int_or_none(image.get('width')),
+                    'height': int_or_none(image.get('height')),
+                })
+
+            entries.append({
+                'id': media_id,
+                'title': title,
+                'description': description,
+                'duration': duration,
+                'creator': creators,
+                'formats': formats,
+                'subtitles': subtitles,
+                'thumbnails': thumbnails,
+            })
+        return entries
+
+    def _json_from_graphql(self, id):
```

Comment:
don't redefine built-ins
```suggestion
    def _json_from_graphql(self, media_id):
```